src/opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -529,25 +529,37 @@ namespace
 /// \brief Check whether each MS well has COMPSEGS entry andissue error if not.
 /// \param welsegs All wells with a WELSEGS entry together with the location.
 /// \param compegs All wells with a COMPSEGS entry
-void check_compsegs_consistency(::Opm::Schedule::WelSegsSet& welsegs, std::set<std::string>&  compsegs)
+void check_compsegs_consistency(::Opm::Schedule::WelSegsSet& welsegs, 
+                                std::set<std::string>&  compsegs,
+                                const std::vector<::Opm::Well>& wells)
 {
     std::vector<std::pair<std::string,::Opm::KeywordLocation>> difference;
     difference.reserve(welsegs.size());
     std::set_difference(welsegs.begin(), welsegs.end(),
                         compsegs.begin(), compsegs.end(),
                         std::back_inserter(difference),
                         ::Opm::Schedule::PairComp());
+    // Ignore wells without connections
+    const auto empty_conn = [wells](const std::pair<std::string,::Opm::KeywordLocation> &x) -> bool {
+        const auto& wname = x.first;
+        const auto it = std::find_if(wells.begin(), wells.end(),
+                                     [wname](const ::Opm::Well& well) -> bool {
+                                         return well.name()==wname && well.getConnections().empty();});
+        return it != wells.end();
```

Comment:
For what it's worth, the sequence `it = find_if(); return it != end()` here can also be written in terms of [`std::any_of()`](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of) as
```C++
return std::any_of(wells.begin(), wells.end(), [wname = x.first](const ::Opm::Well& well)
    { return (well.name() == wname) && well.getConnections().empty(); });
```

Reply:
Nice! Will update!